lib/engine-native/src/serialize.rs

```diff
@@ -33,32 +37,22 @@ pub struct ModuleMetadata {
     pub compile_info: CompileModuleInfo,
     pub prefix: String,
     pub data_initializers: Box<[OwnedDataInitializer]>,
-    // The function body lengths (used to find function by address)
-    pub function_body_lengths: PrimaryMap<LocalFunctionIndex, u64>,
+    pub frame_infos: PrimaryMap<LocalFunctionIndex, CompiledFunctionFrameInfo>,
 }
 
-pub struct ModuleMetadataSymbolRegistry<'a> {
-    pub prefix: &'a String,
+#[derive(MemoryUsage)]
+pub struct ModuleMetadataSymbolRegistry {
+    pub prefix: String,
 }
 
 impl ModuleMetadata {
-    pub fn split<'a>(
-        &'a mut self,
-    ) -> (&'a mut CompileModuleInfo, ModuleMetadataSymbolRegistry<'a>) {
-        let compile_info = &mut self.compile_info;
-        let symbol_registry = ModuleMetadataSymbolRegistry {
-            prefix: &self.prefix,
-        };
-        (compile_info, symbol_registry)
-    }
-
-    pub fn get_symbol_registry<'a>(&'a self) -> ModuleMetadataSymbolRegistry<'a> {
+    pub fn get_symbol_registry(&self) -> ModuleMetadataSymbolRegistry {
         ModuleMetadataSymbolRegistry {
-            prefix: &self.prefix,
+            prefix: self.prefix.clone(),
```

Comment:
Could we not clone this? Can we return a `&ModuleMetadataSymbolRegistry` instead?

Reply:
Cloning it makes things easier to reason about, and the code simpler. I think is a bit better tradeoff?